src/entity/schema.ts

```diff
@@ -0,0 +1,389 @@
+        name: 'blue nominate'
+    },
+    YELLOW_NOMINATE: {
+
+        type: CardType.NOMINATE,
+        colors: [ CardColor.YELLOW ] as readonly CardColor[],
+        name: 'yellow nominate'
+    },
+    MULTI_NOMINATE: {
+        type: CardType.NOMINATE,
+        colors: [ CardColor.RED, CardColor.GREEN, CardColor.BLUE, CardColor.YELLOW ] as readonly CardColor[],
+        name: 'multi nominate'
+    }
+} as const;
+
+export type Card = ObjectValues<typeof Card>;
+
+export const CardSchema = z.object({
+    type: z.string(),
+    colors: z.array(z.nativeEnum(CardColor)),
+    name: z.string(),
+    value: z.number().int().positive().optional()
+});
+
+export type CardSchema = z.infer<typeof CardSchema>;
+
+// ### User ###
+export const UserSchema = z.object({
+    username: z.string()
+});
+
+export type UserSchema = z.infer<typeof UserSchema>;
+
+// ### Player ###
+export const PlayerSchema = UserSchema.extend({
+    socketId: z.string().optional(),
+    cardAmount: z.number().int().positive(),
+    cards: z.array(CardSchema).optional(),
+    ranking: z.number().int().optional(),
+    disqualified: z.boolean()
+});
+
+export type PlayerSchema = z.infer<typeof PlayerSchema>;
+
+export const PlayerInvitationSchema = PlayerSchema.extend({
+    accepted: z.boolean()
+});
+
+export type PlayerInvitationSchema = z.infer<typeof PlayerInvitationSchema>;
+
+// ### Action ###
+export enum ActionType {
+    TAKE = 'take',
+    DISCARD = 'discard',
+    NOMINATE = 'nominate',
+    NOPE = 'nope',
+    DISQUALIFIED = 'disqualified'
+}
+
+export const ActionSchema = z.object({
+    type: z.nativeEnum(ActionType),
+    explanation: z.string(),
+    player: PlayerSchema,
+    cardAmount: z.number().int().positive(),
+    cards: z.array(CardSchema).optional(),
+    nominatedPlayer: PlayerSchema.optional(),
+    nominatedColor: z.nativeEnum(CardColor).optional()
+});
+
+export type ActionSchema = z.infer<typeof ActionSchema>;
+
+// ### Tournament ###
+export enum TournamentState {
+    PREPARATION,
+    ONGOING,
+    FINISHED
+}
+
+export const TournamentSchema = z.object({
+    TODO: z.string()
+});
+
+export type TournamentSchema = z.infer<typeof TournamentSchema>;
+
+// ### Game ###
+export enum GameRole {
+    ROUND_ROBIN = 'round_robin',
+    SUDDEN_DEATH = 'sudden_death'
+}
+
+export enum GameState {
+    GAME_START = 'game_start',
+    NOMINATE_FLIPPED = 'nominate_flipped',
+    TURN_START = 'turn_start',
+    CARD_DRAWN = 'card_drawn',
+    GAME_END = 'game_end',
+    CANCELLED = 'cancelled'
+}
+
+export const GameInvitationSchema = z.object({
+    id: z.string().uuid(),
+    state: z.nativeEnum(GameState),
+    noActionCards: z.boolean(),
+    noWildCards: z.boolean(),
+    oneMoreStartCard: z.boolean(),
+    tournament: TournamentSchema.optional(),
+    gameRole: z.nativeEnum(GameRole).optional(),
+    encouterRound: z.number().int().positive().optional(),
+    players: z.array(PlayerSchema)

```

Comment:
Anzahl der Spieler gemäß Spielregeln
```suggestion
    players: z.array(PlayerSchema).min(2).max(6)
```